app/views/core/CreateAccountModal/BasicInfoView.coffee

```diff
@@ -204,11 +206,11 @@ module.exports = class BasicInfoView extends CocoView
 
     @displayFormSubmitting()
     AbortError = new Error()
-    
+
     @checkEmail()
     .then @checkName()
     .then =>
-      if not (@state.get('checkEmailState') in ['available', 'standby'] and @state.get('checkNameState') is 'available')
+      if not (@state.get('checkEmailState') in ['available', 'standby'] and (@state.get('checkNameState') is 'available') or @signupState.get('path') is 'teacher')
```

Comment:
I think this logic could use some parens or something to clarify what's going on. The whole expression is "not", right? As in abort unless these two states are good or we're a teacher?